src/sentry/profiles/task.py

```diff
@@ -62,13 +62,20 @@ def process_profile_task(
         message_dict = msgpack.unpackb(payload, use_list=False)
         profile = json.loads(message_dict["payload"], use_rapid_json=True)
 
+        if (
+            not message_dict.get("sampled", True)
+            and options.get("profiling.profile_metrics.unsampled_profiles.sample_rate") == 0.0
```

Comment:
did you mean to compare the sample rate to a random number here? Also, why do you read the option in the consumer as well as in the celery task?